packages/sdk/src/authentication/ApiKeyProvider.ts

```diff
@@ -0,0 +1,22 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+import { AxiosRequestConfig } from "axios";
+import { IAuthProvider } from "./IAuthProvider";
+
+export class ApiKeyProvider implements IAuthProvider {
+  private apiKey: string;
+
+  constructor(apiKey: string) {
```

Comment:
the api key may not only added to header, and the name of header may be different for different APIs. We need to expose more parameters.

Reply:
Fixed.